pairing/src/bls12_381/mod.rs

```diff
@@ -23,14 +23,184 @@ pub use self::fr::{Fr, FrRepr};
+        self.0.fmt(f)
+    }
+}
+
+impl ConditionallySelectable for Gt {
+    fn conditional_select(a: &Self, b: &Self, choice: Choice) -> Self {
+        Gt(Fq12::conditional_select(&a.0, &b.0, choice))
+    }
+}
+
+impl Neg for Gt {
+    type Output = Gt;
+
+    fn neg(self) -> Self::Output {
+        let mut ret = self.0;
+        ret.conjugate();
+        Gt(ret)
+    }
+}
+
+impl Sum for Gt {
+    fn sum<I: Iterator<Item = Self>>(iter: I) -> Self {
+        iter.fold(Self::identity(), Add::add)
+    }
+}
+
+impl<'r> Sum<&'r Gt> for Gt {
+    fn sum<I: Iterator<Item = &'r Self>>(iter: I) -> Self {
+        iter.fold(Self::identity(), Add::add)
+    }
+}
+
+impl Add for Gt {
+    type Output = Gt;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        Gt(self.0 * rhs.0)
+    }
+}
+
+impl Add<&Gt> for Gt {
+    type Output = Gt;
+
+    fn add(self, rhs: &Gt) -> Self::Output {
+        Gt(self.0 * rhs.0)
+    }
+}
+
+impl AddAssign for Gt {
+    fn add_assign(&mut self, rhs: Self) {
+        self.0 *= rhs.0;
+    }
+}
+
+impl AddAssign<&Gt> for Gt {
+    fn add_assign(&mut self, rhs: &Gt) {
+        self.0 *= rhs.0;
+    }
+}
+
+impl Sub for Gt {
+    type Output = Gt;
+
+    fn sub(self, rhs: Self) -> Self::Output {
+        self + (-rhs)
+    }
+}
+
+impl Sub<&Gt> for Gt {
+    type Output = Gt;
+
+    fn sub(self, rhs: &Gt) -> Self::Output {
+        self + (-*rhs)
+    }
+}
+
+impl SubAssign for Gt {
+    fn sub_assign(&mut self, rhs: Self) {
+        *self = *self - rhs;
+    }
+}
+
+impl SubAssign<&Gt> for Gt {
+    fn sub_assign(&mut self, rhs: &Gt) {
+        *self = *self - rhs;
+    }
+}
+
+impl Mul<&Fr> for Gt {
+    type Output = Gt;
+
+    fn mul(self, other: &Fr) -> Self::Output {
+        let mut acc = Self::identity();
+
+        // This is a simple double-and-add implementation of group element
+        // multiplication, moving from most significant to least
+        // significant bit of the scalar.
+        //
+        // We skip the leading bit because it's always unset for Fr
+        // elements.
+        for bit in other
+            .to_repr()
+            .as_ref()
+            .iter()
+            .rev()
+            .flat_map(|byte| (0..8).rev().map(move |i| Choice::from((byte >> i) & 1u8)))
+            .skip(1)
+        {
+            acc = acc.double();
+            acc = Gt::conditional_select(&acc, &(acc + self), bit);
+        }
+
+        acc
+    }
+}
+
+impl Mul<Fr> for Gt {
+    type Output = Gt;
+
+    fn mul(self, other: Fr) -> Self::Output {
+        self * &other
+    }
+}
+
+impl<'r> MulAssign<&'r Fr> for Gt {
+    fn mul_assign(&mut self, other: &'r Fr) {
+        *self = *self * other
+    }
+}
+
+impl MulAssign<Fr> for Gt {
+    fn mul_assign(&mut self, other: Fr) {
+        self.mul_assign(&other);
+    }
+}
+
+impl Group for Gt {
+    type Scalar = Fr;
+
+    fn random<R: RngCore + ?Sized>(_rng: &mut R) -> Self {
+        unimplemented!()
```

Comment:
Unimplemented because we don't use this API in our stack, and will be deleting all this code soon.